### PR TITLE
Export

### LINKED ISSUE #97: Tables - add option for export functionality
I'm thinking we can start by supporting `settings.export: ["csv", "xlsx"]` (default: null)

### PR DESCRIPTION
**Pull request by samussiah**
Closes #97

### CODE REVIEW

**Comment by samussiah** [comment]
Reason for failed check: https://stackoverflow.com/a/14737423/4142034

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
   "main": "lib/webcharts.common.js",
   "jsnext:main": "src/index.js",
   "dependencies": {
-    "d3": "^3"
+    "d3": "^3",
+    "js-xlsx": "^0.8.22"
```
Comment (jwildfire) [inline package.json]: I'm thinking we should move this to [`optionalDependencies`](https://docs.npmjs.com/files/package.json#optionaldependencies)

File: src/table/draw.js [hunk 2]
```diff
@@ -137,7 +138,13 @@ export default function draw(passed_data, processed_data) {
     }
 
     //Add sort.
-    if (this.config.sort) this.sort.addSort.call(this);
+    if (this.config.sortable) this.sortable.addSort.call(this);
+
+    //Add pagination.
```
Comment (jwildfire) [inline src/table/draw.js]: fix comment

File: src/table/exportable/exports/csv.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        });
+
+        CSVarray.push(row);
+    });
+
+    //transform CSV array into CSV string
+    const CSV = new Blob([CSVarray.join('\n')], { type: 'text/csv;charset=utf-8;' }),
+        fileName = `Selected Data.csv`,
+        link = this.wrap.select('.export#csv');
+
```
Comment (jwildfire) [inline src/table/exportable/exports/csv.js]: What is navigator?

File: src/table/exportable/exports/csv.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            return `"${value}"`;
+        });
+
+        CSVarray.push(row);
+    });
+
+    //transform CSV array into CSV string
+    const CSV = new Blob([CSVarray.join('\n')], { type: 'text/csv;charset=utf-8;' }),
+        fileName = `Selected Data.csv`,
```
Comment (jwildfire) [inline src/table/exportable/exports/csv.js]: Let's avoid spaces in file names, and I tend to like filenames more descriptive. How about "webchartsTablesExport_yyyymmdd_hhmmss.csv".

File: src/table/exportable/exports/xlsx.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                view = new Uint8Array(buffer);
+
+            for (let i = 0; i !== s.length; ++i) view[i] = s.charCodeAt(i) & 0xff;
+
+            return buffer;
+        }; // convert spreadsheet to binary or something, i don't know
+
+    //transform CSV array into CSV string
+    const blob = new Blob([s2ab(xlsx)], { type: 'application/octet-stream;' }),
+        fileName = `Selected Data.xlsx`,
```
Comment (jwildfire) [inline src/table/exportable/exports/xlsx.js]: same comment as above

**Comment by jwildfire** [comment]
Need to get the tests passing somehow ... can we just set `exportable: false` in the config tests?

File: src/table/exportable/exports/csv.js (hunk 3 above)
Comment (samussiah) [reply src/table/exportable/exports/csv.js]: https://www.w3schools.com/jsref/obj_navigator.asp

**Comment by samussiah** [comment]
Changes incorporated in data refactor ( #125 ).  Closing this PR.

**Pull request closed without merging**